ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -197,17 +202,29 @@ object Update {
 
   /** Signal that a command submitted via [[WriteService]] was rejected.
     *
-    * See the different [[RejectionReason]] for why a command can be
-    * rejected.
+    * TODO(v2) add commentary for all parameters, and in particular, refer to the guidance on the error messages given via the Ledger API
     */
   final case class CommandRejected(
       recordTime: Timestamp,
       submitterInfo: SubmitterInfo,
-      reason: RejectionReason,
+      reason: com.google.rpc.status.Status,
```

Comment:
@andreaslochbihler-da : for `KVUtils` (and likely also for Daml on Corda), it would be great to have the help of the `IndexDB` for resolving `(submissionId, submissionRank)` tuples to their completion offset. With the parallel indexer, this actually means that one needs the help of the `Indexer`, as there is an unspecified number of completions that the indexer is in process of persisting; and lookups need to be resolved against them as well as historic completions.

My hunch is that delegating that work explicitly to the indexer in a typed fashion (i.e., a Sum-type here) would lead to a cleaner design, than trying to wire it up via method calls from the `ReadService` implementation to the `Indexer`

Reply:
I'll give it a try.